commonknowledge/meep-intelligence-hub
# Clustering member data by political geographies

This PR provides statistical data on member lists at the regional / constituency / council / ward levels, dependent on the user's current zoom, so that we are always providing some kind of insight for the current view.

As part of this work, the codebase now ingests Region, Constituency and Ward geographies for the whole of the UK, and in PostGIS format unlike before.

The scene is set for geography explorer panels and other interactive map features.

<img width="1055" alt="Screenshot 2024-03-19 at 18 24 29" src="https://github.com/commonknowledge/meep-intelligence-hub/assets/237556/2fbd691c-dafb-43a2-8f48-51078d07c7d0">
<img width="1360" alt="Screenshot 2024-03-19 at 18 24 36" src="https://github.com/commonknowledge/meep-intelligence-hub/assets/237556/f2a4fb85-2280-48a8-b8d3-7e751104c8a8">
<img width="1365" alt="Screenshot 2024-03-19 at 18 24 41" src="https://github.com/commonknowledge/meep-intelligence-hub/assets/237556/7fbd8cb1-bd14-4d50-ba43-340840f308a6">

## Motivation

Displayed member data was previously scattered across the map like a grain truck that's tipped over onto the road. 

<img width="737" alt="Screenshot 2024-03-18 at 18 47 05" src="https://github.com/commonknowledge/meep-intelligence-hub/assets/237556/751b27e0-9e2d-4015-ad21-a96849292936">

## How Can It Be Tested?

1. Re-build the dev container if you haven't already, as we've switched to PostGIS
2. Import the required geo data:
    - `python manage.py import_areas`
    - `python manage.py import_regions`
    - `python manage.py import_wards`
3. Connect a data source of the type `Member` - make sure it has postcodes!
4. Make a report and add a map layer via the `data configuration` option in the top left
5. As you zoom far to near, you should see geographical regions and their aggregate counts.
6. At the nearest zoom level, individual member pins should be displaying.

You can also run the import tests to verify that data is indeed importing.

## Screenshots (if appropriate):

## Types of changes
